dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1029 @@
+from typing import Dict, List, Optional, TYPE_CHECKING, Any, Union
+
+from libcloudforensics.providers.gcp.internal import common as gcp_common
+from libcloudforensics.providers.gcp.internal import log as gcp_log
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+
+from dftimewolf.lib.processors import gcp_cloud_resource_tree_helper as gcp_crt_helper  # pylint: disable=line-too-long
+
+if TYPE_CHECKING:
+  from dftimewolf.lib import state
+
+class GCPCloudResourceTree(module.BaseModule):
+  """GCP Cloud Resource Tree Creator.
+
+  Attributes:
+    project_id (str): Id of the project where the resource is located.
+    zone (str): zone where the resource is located.
+    resource_id (str): Id of the the resource to build the tree for.
+    resource_name (str): Name of the resource to build the tree for.
+    resource_type (str): Resource type.
+    mode (gcp_crt_helper.OperatingMode): Module operation mode
+    resources_dict (Dict[str, Resource]): Dictionary of resources
+    period_covered_by_retrieved_logs (Dict[str, datetime]): Dictionary of the
+        time range of logs retrieved and processed
+
+  """
+
+  def __init__(self,
+               state: 'state.DFTimewolfState',
+               name: Optional[str] = None,
+               critical: bool = True) -> None:
+    """Initializes the Cloud Resource Tree Processor.
+
+    Args:
+      state: recipe state.
+      name: The module's runtime name.
+      critical: True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super(GCPCloudResourceTree, self).__init__(state,
+                                               name=name,
+                                               critical=critical)
+
+    self.project_id: Optional[str] = None
+    self.zone: Optional[str] = None
+    self.resource_id: str = str()
+    self.resource_name: str = str()
+    self.resource_type: Optional[str] = None
+    self.mode: gcp_crt_helper.OperatingMode = gcp_crt_helper.OperatingMode.ONLINE  # pylint: disable=line-too-long
+    self.period_covered_by_retrieved_logs: Dict[str, datetime] = {}
+    self.resources_dict: Dict[str, gcp_crt_helper.Resource] = {}
+
+  # pylint: disable=arguments-differ
+  def SetUp(self, project_id: str, zone: str, resource_id: str,
+            resource_name: str, resource_type: str, mode: str) -> None:
+    """Sets up the resource we want to build the tree for.
+
+    Args:
+      project_id: Project id where the resource are located.
+      zone: zone where the resource are located.
+      resource_id: Resource id.
+      resource_name: Resource name.
+      resource_type: Resource type (currently supported types: gce_instance,
+          gce_disk, gce_image, gce_machine_image, gce_instance_template,
+          gce_snapshot)
+      mode: operational mode (online or offline)
+    """
```

Comment:
Making resource_id and resource_name optional as these can be None
```suggestion
  def SetUp(self,
                     project_id: str, zone: str,
                     resource_type: str,
                     mode: str,
                     resource_id: Optional[str] = None,
                     resource_name: Optional[str] = None) -> None:
    """Sets up the resource we want to build the tree for.

    Args:
      project_id: Project id where the resource are located.
      zone: zone where the resource are located.
      resource_type: Resource type (currently supported types: gce_instance,
          gce_disk, gce_image, gce_machine_image, gce_instance_template,
          gce_snapshot)
      mode: operational mode (online or offline).
      resource_id: Resource id.
      resource_name: Resource name.
    """
```